tests/lib/rules/no-extra-parens.js

```diff
@@ -3444,6 +3444,22 @@ ruleTester.run("no-extra-parens", rule, {
                 `a ${operator} function () {};`,
                 "Identifier"
             )
-        )
+        ),
+
+        // Potential directives (no autofix)
+        invalid("('use strict');", null),
+        invalid("function f() { ('abc'); }", null),
+        invalid("(function () { ('abc'); })();", null),
+        invalid("_ = () => { ('abc'); };", null),
+        invalid("'use strict';(\"foobar\");", null),
```

Comment:
I think these all would really become directives. It would be good to also add a test where the expression statement wouldn't end up in the directive prologue so that we confirm that the no-autofix behavior in such case is intentional.

For example: `invalid("foo(); ('bar');", null),`